tests/drivers/adc/adc_api/src/test_adc.c

```diff
@@ -29,7 +29,7 @@
 #elif defined(CONFIG_BOARD_NRF51DK_NRF51422)
 
 #include <hal/nrf_adc.h>
-#define ADC_DEVICE_NAME		DT_ADC_0_NAME
+#define ADC_DEVICE_NAME		DT_LABEL(DT_INST(0, nordic_nrf_adc))
```

Comment:
How can we be sure that `DT_INST(0, nordic_nrf_adc)` is `ADC_0`? Shouldn't we use `DT_NODELABEL()` here?

Reply:
I'm guessing in this case its know that ADC is a singleton, since the driver uses DT_INST.

Reply:
It was more of a general comment to the conversions performed in this PR. Some of the coversions convert from ADC_1 to DT_INST(0, ...).

Reply:
If the driver is multi-instance at all, the INST ordering should be assumed to be basically random, so this is indeed an issue in general (though for this particular line of source it's a singleton).

Reply:
@mbolivar-nordic I might have been a bit too trigger-happy with this one. @henrikbrixandersen please feel free to continue the conversation here and then we can create a separate PR if necessary to tweak whatever is needed.

Reply:
I personally think the INST conversions should be removed and replaced with either NODELABEL or ALIAS.

Reply:
I agree.

Reply:
I'd prefer NODELABEL

Reply:
or in this case, for a lot of these PATH would make sense since they are board specific ifdefs.

Reply:
Either NODELABEL or PATH seems fine as way to resolve this issue